File: demo.in-jpaentity-med-validation/src/main/kotlin/com/example/demo/user_interface/constraints/UniqueSeteNummerConstraintValidator.java
package com.example.demo.user_interface.constraints;

import com.example.demo.user_interface.Sete;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;
import java.util.List;

public class UniqueSeteNummerConstraintValidator implements ConstraintValidator<UniqueSeteNummer, List<Sete>> {
    @Override
    public boolean isValid(List<Sete> values, ConstraintValidatorContext context) {
        boolean duplicate = values.stream()
                .anyMatch(sete1 ->
                        values.stream()
                                .anyMatch(sete2 -> sete1 != sete2 && sete1.getNummer() == sete2.getNummer())
                );
        return !duplicate;
    }
}

File: in-jpaentity-with-validation/src/main/kotlin/com/example/in_jpaentity_with_validation/user_interface/constraints/UniqueSeteNummerConstraintValidator.kt
package com.example.in_jpaentity_with_validation.user_interface.constraints

import com.example.in_jpaentity_with_validation.user_interface.Sete
import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

class UniqueSeteNummerConstraintValidator : ConstraintValidator<UniqueSeteNummer?, List<Sete>> {
    override fun isValid(values: List<Sete>, context: ConstraintValidatorContext): Boolean {
        val duplicate = values.stream()
                .anyMatch { sete1: Sete ->
                    values.stream()
                            .anyMatch { sete2: Sete -> sete1 !== sete2 && sete1.nummer == sete2.nummer }
                }
        return !duplicate
    }
}
